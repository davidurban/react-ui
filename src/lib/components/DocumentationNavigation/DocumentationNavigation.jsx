import PropTypes from 'prop-types';
import React from 'react';
import styles from './DocumentationNavigation.scss';

const DocumentationNavigation = props => (
  <ul className={styles.root}>
    {props.items.map(item => (
      <li key={item.link}>
        <a href={item.link} className={styles.navigationLink}>{item.title}</a>
        {item.items && (
          <ul className={styles.subnavigation}>
            {item.items.map(nestedItem => (
              <li key={nestedItem.link}>
                <a href={nestedItem.link} className={styles.subnavigationLink}>
                  {nestedItem.title}
                </a>
              </li>
            ))}
          </ul>
        )}
      </li>
    ))}
  </ul>
);

DocumentationNavigation.propTypes = {
  items: PropTypes.arrayOf(PropTypes.shape({
    items: PropTypes.arrayOf(PropTypes.shape({
      link: PropTypes.string.isRequired,
      title: PropTypes.string.isRequired,
    })),
    link: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
  })).isRequired,
};

export default DocumentationNavigation;
