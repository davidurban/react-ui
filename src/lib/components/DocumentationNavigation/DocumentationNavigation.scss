@import '../../shared/settings/colors';
@import '../../shared/settings/offsets';
@import '../../shared/settings/typography';
@import '../../shared/tools/spacing';

.headline {
  padding: map-get($offset-values, 6) map-get($offset-values, 5) map-get($offset-values, 2);
  font-weight: 400;
  font-size: $typography-size-3;
}

.root,
.subnavigation {
  padding-left: 0;
  list-style: none;
}

.navigationLink,
.subnavigationLink {
  display: block;
  padding: map-get($offset-values, 3) map-get($offset-values, 5);
  text-decoration: none;
  color: inherit;
}

.navigationLink {
  font-weight: bold;
}

.subnavigationLink {
  padding-left: map-get($offset-values, 7);
  color: map-get($colors-grays, gray-600);
}

.navigationLink:hover,
.navigationLink:focus,
.subnavigationLink:hover,
.subnavigationLink:focus {
  text-decoration: none;
  color: inherit;
  background-color: map-get($colors-grays, gray-50);
}
