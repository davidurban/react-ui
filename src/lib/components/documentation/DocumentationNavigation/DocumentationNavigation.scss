@import '../../../styles/settings/colors';
@import '../../../styles/settings/offsets';
@import '../../../styles/settings/typography';
@import '../../../styles/tools/spacing';
@import '../../../styles/tools/transitions';

.headline {
  padding: map-get($offset-values, 6) map-get($offset-values, 5) map-get($offset-values, 2);
  font-weight: $typography-font-weight-regular;
  font-size: $typography-size-3;
}

.root,
.subnavigation {
  padding-left: 0;
  list-style: none;
}

.navigationLink,
.subnavigationLink {
  @include transition((color, background));

  display: block;
  padding: map-get($offset-values, 3) map-get($offset-values, 5);
  text-decoration: none;
  color: inherit;
}

.navigationLink {
  font-weight: $typography-font-weight-bold;
}

.subnavigationLink {
  padding-left: map-get($offset-values, 7);
  color: map-get($colors-grays, gray-600);
}

.navigationLink:hover,
.navigationLink:focus,
.subnavigationLink:hover,
.subnavigationLink:focus {
  text-decoration: none;
  color: inherit;
  background-color: map-get($colors-grays, gray-50);
}
