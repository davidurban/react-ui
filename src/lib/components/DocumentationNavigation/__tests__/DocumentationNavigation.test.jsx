import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import DocumentationNavigation from '..';

describe('rendering', () => {
  const navigationTreeSingleItem = [
    {
      link: '#anchor',
      title: 'Single item',
    },
  ];
  const navigationTreeMultipleItems = [
    {
      link: '#anchor1',
      title: 'Multiple items 1',
    },
    {
      link: '#anchor2',
      title: 'Multiple items 2',
    },
    {
      link: '#anchor3',
      title: 'Multiple items 3',
    },
  ];
  const navigationTreeNestedItems = [
    {
      items: [
        {
          link: '#anchor1a',
          title: 'Nested items 1 a',
        },
        {
          link: '#anchor1b',
          title: 'Nested items 1 b',
        },
      ],
      link: '#anchor1',
      title: 'Nested items 1',
    },
    {
      link: '#anchor2',
      title: 'Nested items 2',
    },
    {
      link: '#anchor3',
      title: 'Nested items 3',
    },
  ];

  it('renders correctly with single item', () => {
    const tree = shallow(<DocumentationNavigation tree={navigationTreeSingleItem} />);

    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it('renders correctly with multiple items', () => {
    const tree = shallow(<DocumentationNavigation tree={navigationTreeMultipleItems} />);

    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it('renders correctly with nested items', () => {
    const tree = shallow(<DocumentationNavigation tree={navigationTreeNestedItems} />);

    expect(shallowToJson(tree)).toMatchSnapshot();
  });
});
