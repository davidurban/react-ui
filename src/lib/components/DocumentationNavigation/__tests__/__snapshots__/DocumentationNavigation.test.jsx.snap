// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering renders correctly with multiple items 1`] = `
<div>
  <h1
    className="headline"
  >
    React UI
  </h1>
  <ul
    className="root"
  >
    <li
      key="#anchor1"
    >
      <a
        className="navigationLink"
        href="#anchor1"
      >
        Multiple items 1
      </a>
    </li>
    <li
      key="#anchor2"
    >
      <a
        className="navigationLink"
        href="#anchor2"
      >
        Multiple items 2
      </a>
    </li>
    <li
      key="#anchor3"
    >
      <a
        className="navigationLink"
        href="#anchor3"
      >
        Multiple items 3
      </a>
    </li>
  </ul>
</div>
`;

exports[`rendering renders correctly with nested items 1`] = `
<div>
  <h1
    className="headline"
  >
    React UI
  </h1>
  <ul
    className="root"
  >
    <li
      key="#anchor1"
    >
      <a
        className="navigationLink"
        href="#anchor1"
      >
        Nested items 1
      </a>
      <ul
        className="subnavigation"
      >
        <li
          key="#anchor1a"
        >
          <a
            className="subnavigationLink"
            href="#anchor1a"
          >
            Nested items 1 a
          </a>
        </li>
        <li
          key="#anchor1b"
        >
          <a
            className="subnavigationLink"
            href="#anchor1b"
          >
            Nested items 1 b
          </a>
        </li>
      </ul>
    </li>
    <li
      key="#anchor2"
    >
      <a
        className="navigationLink"
        href="#anchor2"
      >
        Nested items 2
      </a>
    </li>
    <li
      key="#anchor3"
    >
      <a
        className="navigationLink"
        href="#anchor3"
      >
        Nested items 3
      </a>
    </li>
  </ul>
</div>
`;

exports[`rendering renders correctly with single item 1`] = `
<div>
  <h1
    className="headline"
  >
    React UI
  </h1>
  <ul
    className="root"
  >
    <li
      key="#anchor"
    >
      <a
        className="navigationLink"
        href="#anchor"
      >
        Single item
      </a>
    </li>
  </ul>
</div>
`;
